vendor/k8s.io/kubernetes/pkg/kubectl/cmd/drain.go

```diff
@@ -208,27 +222,47 @@ func (o *DrainOptions) SetupDrain(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	o.nodeInfos = []*resource.Info{}
 	o.mapper, o.typer = o.Factory.Object()
 
 	cmdNamespace, _, err := o.Factory.DefaultNamespace()
 	if err != nil {
 		return err
 	}
 
-	r := o.Factory.NewBuilder(true).
+	nameArgs := []string{"nodes"}
+
+	b := o.Factory.NewBuilder(true).
 		NamespaceParam(cmdNamespace).DefaultNamespace().
-		ResourceNames("node", args[0]).
-		Do()
+		SelectorParam(o.Selector)
+
+	if len(args) > 0 {
```

Comment:
Move that if before you create the builder and you can then squash the whole builder invocation into one.